### DISCUSSION TITLE
Placeholder using JavaScript data

### DISCUSSION THREAD

**Discussion by diderich** [brianvoe/slim-select#414] [Q&A] [answered]
How can I add a placeholder text when the selector is popuplated using JavaScript / `data:` rather than `<option data-placeholder="true"></option>`?

Using `settings: { placeholderText: 'Text' }` does seem to not work!

**Comment by brianvoe** [comment, marked as answer]
```<option data-placeholder="true">Placeholder text here.</option>```

**Comment by diderich** [comment]
_That unfortunately does not work :-(_

My source code:
![ss-src](https://user-images.githubusercontent.com/10230144/210060248-e9a5955c-d579-47d9-8ec6-6f1d50fbaa16.png)

My data:
```JavaScript
new SlimSelect({ select: '#rtc-id-sel-support-language', data: [ 
{ text: 'German (Germany)', value: 'de-DE', selected: false, html: '<span class="fi fi-de"></span>&nbsp;German (Germany)' },
{ text: 'German (Switzerland)', value: 'de-CH', selected: false, html: '<span class="fi fi-ch"></span>&nbsp;German (Switzerland)' },
{ text: 'English (United States)', value: 'en-US', selected: false, html: '<span class="fi fi-us"></span>&nbsp;English (United States)' },
{ text: 'English (United Kingdom)', value: 'en-GB', selected: false, html: '<span class="fi fi-gb"></span>&nbsp;English (United Kingdom)' },
{ text: 'French (France)', value: 'fr-FR', selected: false, html: '<span class="fi fi-fr"></span>&nbsp;French (France)' },
{ text: 'French (Switzerland)', value: 'fr-CH', selected: false, html: '<span class="fi fi-ch"></span>&nbsp;French (Switzerland)' }]});
```

How I see the data in the selector after having instantiated SlimSelect:
![ss-rendered](https://user-images.githubusercontent.com/10230144/210060332-6bbca974-df5b-48e2-87f4-93653b8e4cf3.png)

How the selector data looks like:
![ss-selector](https://user-images.githubusercontent.com/10230144/210060346-d1f69bab-41a8-4352-ba13-379ee09f8671.png)

How it looks on screen:
![closed](https://user-images.githubusercontent.com/10230144/210060592-e5c857cf-7287-4566-b552-307bf0c66c12.png)
![open](https://user-images.githubusercontent.com/10230144/210060597-4035927a-5f56-415e-b718-cc94ea27f6ea.png)

**Comment by brianvoe** [comment]
Looks like your immediately overriding your html select with data that doesnt have a placeholder in it.

**Reply by diderich** [reply]
That's what I also thought. What I have not yet figured out is how to add a placeholder when passing new data using `data: [ ... ]`. 

Adding `setting: { placeholderText: 'xxx' }` when creating the SlimSelect does nothing. 

Is there a way to specific the equivalent of  `<option data-placeholder="true">Placeholder text here.</option>` in the `data: [ ... ]` section?

**Comment by brianvoe** [comment]
placeholder: true

**Comment by brianvoe** [comment]
you can look at the data options in the docs

**Comment by diderich** [comment]
Thanks a lot for your patience and your help. Now I got it. There needs to be an entry in the `data:`  section with `{placeholder: true, text: 'Placeholder text'}`!
